src/main/java/com/beerjournal/breweriana/collection/persistence/UserCollectionRepository.java

```diff
@@ -29,18 +36,35 @@
         return crudRepository.findOneByOwnerId(ownerId);
     }
 
-    public Set<ItemRef> findAllNotInUserCollection(ObjectId ownerId) {
+    public Page<ItemRef> findAllInUserCollection(ObjectId ownerId, int page, int count, String filterVariableName, String filterVariableValue) {
+        UserCollection userCollection = crudRepository.findOneByOwnerId(ownerId)
+                .orElseThrow(() -> new BeerJournalException(USER_COLLECTION_NOT_FOUND));
+        List<ItemRef> list = new ArrayList<>(userCollection.getItemRefs());
+
+        Stream<ItemRef> listStream = Strings.isNullOrEmpty(filterVariableValue) ?
+                list.stream() :
+                list.stream().filter(v -> filterVariableName.equals("name") ?
+                        v.getName().startsWith(filterVariableValue) : v.getType().startsWith(filterVariableValue));
+
+        return new PageImpl<>(
+                listStream
+                        .skip(page * count)
+                        .limit(count)
+                        .collect(Collectors.toList()),
+                new PageRequest(page, count),
+                list.size());
```

Comment:
I believe that there is no need to use crudRepostory (need to query all items to find them for given userId) for this operation and stream on given list is enough. It shouldn't be a problem with filtering. Am I right?